packages/fx-core/src/plugins/resource/function/index.ts

```diff
@@ -101,30 +102,35 @@ export class FunctionPlugin implements Plugin {
         return res;
     }
 
-    private async runWithErrorWrapper(ctx: PluginContext, name: string, fn: () => Promise<FxResult>): Promise<FxResult> {
+    private async runWithErrorWrapper(
+        ctx: PluginContext,
+        name: string,
+        fn: () => Promise<FxResult>,
+        sendTelemetry = true
+    ): Promise<FxResult> {
         try {
-            telemetryHelper.sendStartEvent(ctx, name);
+            sendTelemetry && telemetryHelper.sendStartEvent(ctx, name);
             const res: FxResult = await fn();
-            telemetryHelper.sendResultEvent(ctx, name, res);
+            sendTelemetry && telemetryHelper.sendResultEvent(ctx, name, res);
             return res;
         } catch (e) {
             if (e instanceof UserError || e instanceof SystemError) {
                 const res = err(e);
-                telemetryHelper.sendResultEvent(ctx, name, res);
+                sendTelemetry && telemetryHelper.sendResultEvent(ctx, name, res);
```

Comment:
I think this kind of errors are from other components, do we send telemetry for them?

Reply:
It is not sent for reporting error. It is sent for that the life cycle exits because of this error. we send it for our life cycle. And we can't assume that library always maintain telemetry.

Reply:
Make sense.